src/ui/Omnibox/Omnibox.ts

```diff
@@ -245,6 +245,8 @@ export class Omnibox extends Component {
     const originalValueForQuerySyntax = this.options.enableQuerySyntax;
     this.options = _.extend({}, this.options, this.componentOptionsModel.get(ComponentOptionsModel.attributesEnum.searchBox));
 
+    $$(this.element).toggleClass('coveo-query-syntax-disabled', this.options.enableQuerySyntax == false);
```

Comment:
I believe this would be more resilient if it was a `removeClass` instead.

Reply:
It would have to be "addClass" instead, with a if check on the fact that the option is disabled or not.

It's a bit more concise like this.